src/store/prompts/prompts.epics.ts

```diff
@@ -376,27 +389,38 @@ const publishPromptEpic: AppEpic = (action$, state$) =>
     map(({ payload }) => ({
       publishRequest: payload,
       prompts: PromptsSelectors.selectPrompts(state$.value),
-      publishedFolders: PromptsSelectors.selectSectionFolders(
-        state$.value,
-        PublishedWithMeFilter,
-      ),
+      publishedAndTemporaryFolders:
+        PromptsSelectors.selectTemporaryAndFilteredFolders(
+          state$.value,
+          PublishedWithMeFilter,
+        ),
     })),
-    switchMap(({ publishRequest, prompts, publishedFolders }) => {
+    switchMap(({ publishRequest, prompts, publishedAndTemporaryFolders }) => {
       const sharedPrompts = prompts
         .filter((prompt) => prompt.id === publishRequest.id)
         .map(({ folderId: _, ...prompt }) => ({
           ...prompt,
           ...resetShareEntity,
           id: uuidv4(),
           originalId: prompt.id,
-          folderId: getFolderIdByPath(publishRequest.path, publishedFolders),
+          folderId: getFolderIdByPath(
+            publishRequest.path,
+            publishedAndTemporaryFolders,
+          ),
           publishedWithMe: true,
           name: publishRequest.name,
           publishVersion: publishRequest.version,
           shareUniqueId: publishRequest.shareUniqueId,
         }));
 
+      const temporaryFolders = getTemporaryFoldersToPublish(
+        publishedAndTemporaryFolders,
+        sharedPrompts[sharedPrompts.length - 1].folderId,
```

Comment:
Are you sure that correct folder will always be on last place i.e. the order is correct?